pkg/routeagent_driver/handlers/kubeproxy/gw_transition.go

```diff
@@ -34,7 +35,7 @@ func (kp *SyncHandler) TransitionToNonGateway() error {
 	// If the active Gateway transitions to a new node, we flush the HostNetwork routing table.
 	kp.updateRoutingRulesForHostNetworkSupport(nil, Flush)
 
-	err := kp.configureIPRule(Delete)
+	err := kp.netLink.ConfigureIPRule(netlinkAPI.Delete, constants.RouteAgentHostNetworkTableID)
```

Comment:
With the suggested changes, this becomes:

```suggestion
	err := kp.netLink.RuleDelIfPresent(netlinkAPI. NewTableRule(
	constants.RouteAgentHostNetworkTableID))
```